triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Cli.scala

```diff
@@ -4,43 +4,27 @@
 package com.daml.auth.middleware.oauth2
 
 import akka.http.scaladsl.model.Uri
-import com.auth0.jwt.algorithms.Algorithm
 import com.daml.auth.middleware.oauth2.Config.{
   DefaultCookieSecure,
   DefaultHttpPort,
   DefaultLoginTimeout,
   DefaultMaxLoginRequests,
 }
 import com.daml.cliopts
-import com.daml.jwt.{
-  ECDSAVerifier,
-  HMAC256Verifier,
-  JwksVerifier,
-  JwtVerifierBase,
-  JwtVerifierConfigurationCli,
-  RSA256Verifier,
-}
+import com.daml.jwt.{JwtVerifierBase, JwtVerifierConfigurationCli}
 import com.typesafe.scalalogging.StrictLogging
-import pureconfig.{ConfigReader, ConfigSource, ConvertHelpers}
-import pureconfig.error.ConfigReaderException
+import pureconfig.ConfigSource
+import pureconfig.error.ConfigReaderFailures
 import scopt.OptionParser
 
 import java.io.File
-import pureconfig.generic.semiauto._
-
 import java.nio.file.{Path, Paths}
+
 import scala.concurrent.duration
 import scala.concurrent.duration.FiniteDuration
+import scalaz.syntax.std.option._
 
-sealed trait ConfigError extends Product with Serializable {
-  def msg: String
-}
-case object MissingConfigError extends ConfigError {
-  val msg = "Missing auth middleware config file"
-}
-final case class ConfigParseError(msg: String) extends ConfigError
-
-final case class Cli(
+private[oauth2] final case class Cli(
```

Comment:
It doesn't look like this is a `case class`. There plenty of behavior encapsulated in this, including logging.

Reply:
I think the important question is "identity", not "behavior". In other words, if `expr: C`, and for all `f`, `f(expr) = f(expr.copy())`, then `C` is a `case class`. Less formally, a case class is exactly the sum of its parts, which seems to hold here.

All of these methods can be written as auxiliary functions; their presence in the class body shouldn't change how we interpret the class's semantics.